server/src/routes/ratings/catering/get.rs

```diff
@@ -0,0 +1,130 @@
+use std::{str::FromStr, vec};
+
+use axum::{Extension, Json};
+use axum_test::TestServer;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+use crate::{error::ApiResult, routes::ratings::types::{MealDto, MealRatingDto, MealSubratingDto}, AppState};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct GetRatingReq {
+    pub date: Option<DateTime<Utc>>,
+}
+
+pub async fn get_catering_rating<'a>(
```

Comment:
What do you think about splitting it into controller and logic. Controller would extract data and other stuff, while function with logic would make a db query